libc/calls/fexecve.c

```diff
@@ -47,7 +51,11 @@ int fexecve(int fd, char *const argv[], char *const envp[]) {
   } else {
     STRACE("fexecve(%d, %s, %s) → ...", fd, DescribeStringList(argv),
            DescribeStringList(envp));
-    if (IsLinux()) {
+    if (__isfdkind(fd, kFdZip) &&
```

Comment:
Are you planning to implement execve() too?

Reply:
Didn't even consider that it could be polyfilled with similar implementation to for `fexecve`. Added for elf on Linux and FreeBSD. Added (with fd leaking for APE on Linux).